src/config.ts

```diff
@@ -1,29 +1,14 @@
 import { ethers } from "ethers";
 
+import {
+  MissingSignerConfigError,
+  MissingProviderConfigError,
+  MissingStoreConfigError,
+  MissingFromIdConfigError,
+} from "./core/config/configErrors";
 import { StoreInterface } from "./core/store";
 import { HexString } from "./types/Strings";
 
-export const MissingContract = new Error("Contract was not found");
-export const MissingSigner = new Error("Signer is not set.");
-export const MissingProvider = new Error("Blockchain provider is not set.");
-export const MissingStore = new Error("Store adapter was not found");
-export const MissingUser = new Error("No user id found. Please authenticate a handle.");
-
-export interface Contracts {
```

Comment:
I don't understand the reason for making this generic. It loses specificity and documentation right?

Reply:
The problem with the interface is that TypeScript won't let us index the object with `[]`, since the string may or may not be a valid key on the object at runtime. I did some investigation, and [what I found](https://stackoverflow.com/questions/57350092/string-cant-be-used-to-index-type) indicated that it's actually bad practice to define interfaces which will be accessed dynamically since the interface can't provide any real type safety, just the illusion of it.

An alternative we could try if we really want to preserve the typing here would be to replace the `getContracts()` method with a series of methods for each specific contract, i.e. `getAnnouncerContract()`, `getRegistryContract()`, etc, but I figured that was too big of a change for this PR.

Reply:
**Notes from async discussion:**

- We decided to replace this interface with a series of string types for the various contract names and update the contracts type to use a union of the types.
- Fix has already been pushed.